All rules implemented

breadsp opened the pull request:
Code has been condensed and all rules have been created in this new format
Code has been added to write to file-validator and data-validator tables to update accordingly
JSON mode has been incorporated and tested in leidos

breadsp committed:
All rules implemented

Code has been condensed and all rules have been created in this new format
Code has been added to write to file-validator and data-validator tables to update accordingly
JSON mode has been incorporated and tested in leidos

n2iw commented:
nci-seronet-data-validator.py:

- Line 30-33 and 34-37 are duplicates
- Line 160: parameter s3_client not used
- Line 169: parameters filesdb_conn and s3_client not used

Other than these, it looks good to me!

breadsp committed:
Updated Files

updated validation rules for Assay, Assay_Target and Conirmatory_Test_Results
(these rules will change in future relase but all sql and merge logic is now in place)
fixed issues with Participant ID and biospecimen ID for overall submission to only run when appropriate and use correct files
Fixed JSON mode and tested in the leidos account with expected slack output

breadsp committed:
Merge branch 'main' into data_validation_final

breadsp commented:
updated validation rules for Assay, Assay_Target and Conirmatory_Test_Results
(these rules will change in future relase but all sql and merge logic is now in place)
fixed issues with Participant ID and biospecimen ID for overall submission to only run when appropriate and use correct files
Fixed JSON mode and tested in the leidos account with expected slack output

breadsp committed:
Delete Data_Validation_V6.py
